compiler/rustc_metadata/src/rmeta/mod.rs

```diff
@@ -395,6 +395,8 @@ define_tables! {
     associated_item_or_field_def_ids: Table<DefIndex, LazyArray<DefIndex>>,
     opt_def_kind: Table<DefIndex, DefKind>,
     visibility: Table<DefIndex, LazyValue<ty::Visibility<DefIndex>>>,
+    impl_restriction: Table<DefIndex, LazyValue<ty::Restriction>>,
+    mut_restriction: Table<DefIndex, LazyValue<ty::Restriction>>,
```

Comment:
Could we merge those tables? AFAICT, a definition cannot have both a `mut_restriction` and an `impl_restriction`.

Reply:
I don't understand your example. Could you elaborate a bit? Is this extension to be implemented soon?

Reply:
The example I came up with was not part of the RFC, nor do I intend on proposing it any time soon. It's merely a way to show that restrictions could be reasonably expected to overlap in the future. Another reasonable possibility is splitting mutability and constructability, which was discussed in RFC process.

Another possibility is that the restriction types will be differentiated by a generic, preventing their interoperable use (and thus possible bugs). This was mentioned above and is something I'm attempting to make work.